compiler/src/dmd/inline.d

```diff
@@ -1288,6 +1282,14 @@ public:
 
     override void visit(UnaExp e)
     {
+        if (auto ce = e.isCastExp())
```

Comment:
You should override visit(CastExp) instead of the "dynamic cast" here. Probably in the other case, too.